src/scripts/loot.ts

```diff
@@ -1,97 +1,280 @@
-                            Orion.WaitForTarget(1000) && Orion.CancelTarget();
-                        }
-                    }
-
-                    Scripts.Loot.lootCorpseId(id);
+                    Scripts.Loot.lootCorpseId(id, cut, weapon);
                     Orion.Ignore(id);
                     Orion.Wait(100);
                 }
                 listOfCorpses = Orion.FindType('0x2006', '-1', 'ground', 'fast', 2, 'red');
             }
+
+            const itemsOnGround = Orion.FindList('lootItems', 'ground', 'fast', 4);
+            let serverLagActionsLeft = 4;
+            for (const itemId of itemsOnGround) {
+                Orion.MoveItem(itemId, 0, 'myLootBag');
+                Orion.Wait(serverLagActionsLeft ? 50 : 350);
+                serverLagActionsLeft--;
+            }
         }
 
         /**
-         * Scripts.Loot.lootCorpseId
-         * stability beta
-         *
-         * vylotuje z konkretni mrtvolky veci nastavene v lootItems listu
+         * Loot konkretnej mrtvoly
+         * 
+         * @param corpseId string - serial mrtvoly
+         * @param cut boolean - rezat?
+         * @param weapon boolean - pouzivat zbran na rezanie?
          */
-        static lootCorpseId(id:string) {
+        static lootCorpseId(corpseId: string, cut?: boolean, weapon?: boolean) {
             let serverLagActionsLeft = 4;
-            Orion.OpenContainer(id, 5000, `Container id ${id} not found`);
-            let itemsInCorpse = Orion.FindList('lootItems', id);
+            Orion.OpenContainer(corpseId, 5000, `Container id ${corpseId} not found`);
+
+            const hasLootBag = Orion.Count("0x0E76", "0x049A", corpseId) > 0;
+            if (hasLootBag && cut) {
+                Orion.UseObject('cutWeapon');
+                Orion.WaitForTarget(1000);
+                Orion.TargetObject(corpseId);
+                if (weapon) {
+                    Orion.UseObject('fightWeapon');
+                    Orion.WaitForTarget(1000) && Orion.CancelTarget();
+                }
+                Orion.Wait(250);
+            }
+
+            const itemsInCorpse = Orion.FindList('lootItems', corpseId);
             if (itemsInCorpse.length) {
                 for (const itemId of itemsInCorpse) {
                     // TODO doresit prehazeni veci z hlavniho baglu do myLootBag pri lotovani pytliku
                     Orion.MoveItem(itemId, 0, "myLootBag");
-                    Orion.Wait(serverLagActionsLeft ? 50 : 350);
+                    Orion.Wait(serverLagActionsLeft ? 150 : 500);
                     serverLagActionsLeft--;
                 }
             }
         }
 
-        static lootAllFrom(delay = responseDelay) {
-            Scripts.Utils.targetObjectNotSelf('lootAllContainer', `Target object to loot`);
+        /**
+         * Vrati array serialov veci v zakladnom batohu
+         * 
+         * @returns string[]
+         */
+        private static getBagSnapshot(): string[] {
```

Comment:
V te harving verzi mam v ramci `lootCorpsesAround()` takovy snapshot i s poctem (aby to nehazelo v ramci merge itemu vsechno do toho lootpytle). Realne v dungach nelootim, takze to asi neni potreba resit, jen kdyby jsi to povazoval za neco nezbytneho :)

Reply:
imho je to ďalšia záťaž, ktorá je podľa mňa irelevantná. Keď ideš dungy tak máš väčšinou svoje veci v reff bedničke (regy, gp - lebo asi tých sa to týka ten tvoj comment).

Reply:
jo souhlasim.. Maximalne pokud lootnes bandy tak ti je to hodi do loot pytle pokud je mas v zakladnim baglu :)

Reply:
Kto lootuje bandy má iné problémy ako to, že ich bude mať v základnom bágliku :D